### DISCUSSION TITLE
correct behavior of IMU sensor readings

### DISCUSSION THREAD

**Discussion by zimmy87** [microsoft/AirSim#3827] [Support Q&A]
We are finding that after #3801 was merged, we started seeing #3820. However, we are unsure of the appropriate behavior in this scenario. We've tried removing the gravity component in the IMU reading, theorizing that PX4 might be expecting the resultant acceleration of the vehicle, but that fails a PreFlightCheck in PX4 [here](https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/commander/Arming/PreFlightCheck/checks/accelerometerCheck.cpp#L85) which is checking that the magnitude of the IMU's reported acceleration is between 4.0 and 15.0. This would seem to suggest that PX4 is expecting to see the IMU report an acceleration due to gravity, but the only way we can get PX4 to work is output a vector for gravity pointing upwards (negative in the z-axis). This runs counter to our expectations, especially since Mavlink documentation states the HIL_SENSOR message uses a NED coordinate system [here](https://mavlink.io/en/messages/common.html#HIL_SENSOR), so we would expect gravity should be reported as a vector pointing downwards (positive in the z-axis). Does anybody know what the expected behavior for IMU sensors with regards to reporting an acceleration due to gravity?

@saihv or @lovettchris: do either of you have context for this scenario?

**Comment by lovettchris** [comment]
either way there's no way we've had IMU gravity vector wrong for the last few years so I wouldn't mess with that.
